crates/aim-downloader/src/https.rs

```diff
@@ -130,8 +130,15 @@ impl HTTPSHandler {
             .basic_auth(parsed_address.username, Some(parsed_address.password))
             .send()
             .await
-            .map_err(|_| format!("Failed to GET from {} to {}", &input, &output))
-            .unwrap();
+            .and_then(|r| r.error_for_status())
+            .unwrap_or_else(|e| {
+                eprintln!(
+                    "Failed to download {input}: Server returned {:?}",
+                    e.status().unwrap_or_default()
+                );
```

Comment:
Can this be propagated to application layer (e.g `tabby download` command), thus we can generate better error message there?

Reply:
What additional context would you want to include?